### PR TITLE
MM-68447: Fix MCP tab short-query false positives in agent modal search

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary

Short search strings in the Create Agent modal MCPs tab could keep the embedded **Mattermost** server visible even when the query did not appear in the server name or any tool name. The filter matched **tool descriptions**, which are long prose strings and often contain unrelated two-letter substrings (for example `za`).

This change applies **full** matching (name + description) only when the trimmed query is **at least three characters**. For one- and two-character queries, matching is limited to the **server display name** and **tool names** only, which removes those false positives while preserving expected matches (for example a single `e` still matches **Matt**e**r**most).

Search matches only **admin-enabled** tools (`tool.enabled`): disabled tools are excluded from name and description matching so a server cannot appear solely because of a disabled tool row (see CodeRabbit feedback).

Pure filter logic lives in `webapp/src/components/agents/tabs/mcp_servers_filter.ts`; Jest regression tests are in `mcp_servers_filter.test.ts` (empty vs whitespace, short vs long queries, disabled-tool cases, 3-character boundary).

The `e2e-llmbot-real-apis-citations` CI job failed once with an **Anthropic API health check timeout** before any plugin tests ran (transient upstream/network). Real-API pre-flight checks in `e2e/helpers/api-health-check.ts` now **retry** (up to 3 attempts with backoff) on likely-transient connection errors for both Anthropic and OpenAI.

#### Ticket Link

Jira https://mattermost.atlassian.net/browse/MM-68447

#### Release Note

```release-note
Fixed the MCPs tab search in the agent builder so that very short queries no longer match unrelated text inside tool descriptions and incorrectly show the Mattermost server.
```

<div><a href="https://cursor.com/agents/bc-10246cd8-8318-4ef3-9910-a5cdd9c69766"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-10246cd8-8318-4ef3-9910-a5cdd9c69766"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

## Summary by CodeRabbit

* **New Features**
  * Improved MCP server search: queries are trimmed; 1–2 char queries match server and enabled tool names only, while 3+ char queries also match enabled tool descriptions; disabled tools are excluded.

* **Performance**
  * Search filtering optimized for snappier results.

* **Reliability**
  * Provider health checks now retry with backoff to improve stability and latency reporting.

* **Tests**
  * Added comprehensive tests covering search behaviour and edge cases.

### CODE REVIEW

**Comment by nickmisasi** [comment]
@cursor fix the problem pointed out by CodeRabbitAI and then add tests according to the guidance from mattermost-build

File: e2e/helpers/api-health-check.ts [hunk 1]
```diff
@@ -23,89 +23,145 @@ interface HealthCheckResult {
...
-        };
+            const body = await response.text().catch(() => '');
+            lastError = `HTTP ${response.status}: ${body.substring(0, 200)}`;
+            return {
+                provider: 'Anthropic',
+                model: service.defaultModel,
+                healthy: false,
+                error: lastError,
+                latencyMs: Date.now() - overallStart,
+            };
```
[the remark this answers is not in the document]
Comment (nickmisasi) [reply e2e/helpers/api-health-check.ts:73]: @cursor

**Review by crspeller — APPROVED** [review]
